power-pay-frontend/src/hooks/useStorage.ts

```diff
@@ -0,0 +1,27 @@
+//useState and useEffect hooks from React
+import { useState } from 'react';
+
+//UseStorageProps<T>: An interface defining the props for the useStorage hook
+interface UseStorageProps<T> {
+  initialValue?: T  | undefined;
+}
+
+//useStorage<T = string>: A generic function that accepts a UseStorageProps<T> object.
+
+export function useStorage<T = string>(props: UseStorageProps<T>): [T, React.Dispatch<React.SetStateAction<T>> , () => void] {
+
+  const { initialValue } = props;
+
+  //storedValue: A state variable that stores the value retrieved from the localStorage using the provided key
+
+  const [storedValue, setStoredValue] = useState<T>(initialValue!);
```

Comment:
The whole idea of using the context, is not to use a state anymore. This doesn't make sense here. Why aren't you using the context.